src/figuresClasses.ts

```diff
@@ -1,19 +1,98 @@
 export interface Figure {
-
+  shape: 'triangle' | 'rectangle' | 'circle';
+  getArea(): number;
+  color: 'green' | 'red' | 'blue';
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  side1: number;
+
+  side2: number;
+
+  side3: number;
 
+  constructor(color: 'red' | 'green' | 'blue',
+    side1: number, side2: number,
+    side3: number) {
+    this.color = color;
+    this.side1 = side1;
+    this.side2 = side2;
+    this.side3 = side3;
+
+    if (side1 + side2 <= side3
+      || side1 + side3 <= side2
+      || side2 + side3 <= side1
+      || side1 <= 0
+      || side2 <= 0
+      || side3 <= 0) {
+      throw new Error('side 1, 2, 3 cannot form a triangle');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.side1 + this.side2 + this.side3) / 2;
+
+    return Math.floor(Math.sqrt(s * (s - this.side1)
+    * (s - this.side2) * (s - this.side3)) * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
+
+  color: 'red' | 'green' | 'blue';
+
+  radius: number;
 
+  constructor(color: 'red' | 'green' | 'blue',
+    radius: number) {
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again